### PR TITLE
Remove koji fallback config

### PR DESCRIPTION
**Pull request by SalatskySal**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by MartinBasti** [comment]
```diff
diff --git a/tests/plugins/test_add_filesystem.py b/tests/plugins/test_add_filesystem.py
index 16d2c60..42c6d5b 100644
--- a/tests/plugins/test_add_filesystem.py
+++ b/tests/plugins/test_add_filesystem.py
@@ -180,9 +180,23 @@ def mock_image_build_file(tmpdir, contents=None):
     return file_path
 
 
+def mock_koji_config(workflow):
+    workflow.plugin_workspace.setdefault(ReactorConfigPlugin.key, {})
+    reactor_config = workflow.plugin_workspace[ReactorConfigPlugin.key].get(WORKSPACE_CONF_KEY)
+    if not reactor_config:
+        reactor_config = ReactorConfig({})
+    reactor_config.conf['koji'] = {
+        'auth': {},
+        'hub_url': 'https://koji.example.com/koji_hub',
+        'root_url': 'https://koji.example.com/koji_root/',
+    }
+    workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] = reactor_config
+
+
 def mock_workflow(tmpdir, dockerfile=DEFAULT_DOCKERFILE,
                   scratch=False, for_orchestrator=False):
     workflow = DockerBuildWorkflow('test-image', source=MOCK_SOURCE)
+    mock_koji_config(workflow)
     workflow.user_params['scratch'] = scratch
     mock_source = MockSource(tmpdir)
     setattr(workflow, 'builder', X())
```

I think that this is missing piece. Workflow for plugins that needs koji has to be updated. (Of course without copy paste, this function should be reusable)

**Comment by SalatskySal** [comment]
> /packti copr-build

@MartinBasti Please, fix your command

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 1]
```diff
@@ -57,8 +57,52 @@ def get_value(workflow, name, fallback):
         raise
 
 
-def get_koji(workflow, fallback=NO_FALLBACK):
-    koji_map = get_value(workflow, 'koji', fallback)
+def add_koji_map_in_workflow(workflow, hub_url, root_url=None, reserve_build=None,
```
Comment (rcerven) [inline atomic_reactor/plugins/pre_reactor_config.py:60]: why and what is this? 

this is used just in tests to set koji values to reactor config,
and in tests we usually already passing dict to ReactorConfig, so passing it to function which will set it isn't useful

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_reactor_config.py:60]: :eyes: yup mocking function should be in tests.

@rcerven this mocking has been introduced, because koji attribute is mandatory now, and tests we failing with missing `koji` config key

Comment (SalatskySal) [reply atomic_reactor/plugins/pre_reactor_config.py]: I added this function because some plugins' tests do not configure koji config as usual and I need to manually add all configs to each internal test. So I use this function instead of creating configuration dicts every time.

Comment (SalatskySal) [reply atomic_reactor/plugins/pre_reactor_config.py]: This function has been moved to tests/util.py

**Comment by MartinBasti** [comment]
> Do we need an osbs-docs update to mention that koji configuration is now required in configmap?

That's good idea.  :eyes: @SalatskySal

File: atomic_reactor/plugins/exit_store_metadata_in_osv3.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def set_koji_task_annotations_whitelist(self, annotations):
         koji's configuration to be included in the build_annotations.json file,
         which will be attached in the koji task output.
         """
-        koji_config = get_koji(self.workflow, {})
+        koji_config = get_koji(self.workflow)
```
Comment (lcarva) [inline atomic_reactor/plugins/exit_store_metadata_in_osv3.py:179]: This looks like a change in behavior. Should this be wrapped in `try/except` block?

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 3]
```diff
@@ -475,7 +476,14 @@ class ReactorConfig(object):
     Class to parse the atomic-reactor configuration file
     """
 
-    DEFAULT_CONFIG = {ReactorConfigKeys.VERSION_KEY: 1}
+    DEFAULT_CONFIG = {
+        ReactorConfigKeys.VERSION_KEY: 1,
+        ReactorConfigKeys.KOJI_KEY: {
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_reactor_config.py:481]: Why is this needed?

File: atomic_reactor/plugins/pre_resolve_composes.py [hunk 4]
```diff
@@ -77,14 +72,8 @@ def __init__(self, tasker, workflow,
...
         self.koji_target = koji_target
-        self.koji_fallback = {
-            'hub_url': koji_hub,
-            'auth': {
-                'ssl_certs_dir': koji_ssl_certs_dir
-            }
-        }
         if koji_target:
-            if not get_koji(self.workflow, self.koji_fallback)['hub_url']:
+            if not get_koji(self.workflow).get('hub_url'):
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_resolve_composes.py:76]: `hub_url` is a required attribute of the `koji` Json Object. Shouldn't this be `['hub_url']` instead?

File: atomic_reactor/schemas/config.json [hunk 5]
```diff
@@ -791,5 +791,5 @@
         "required": ["url"]
     }
   },
-  "required": ["version"]
+  "required": ["version", "koji"]
```
Comment (lcarva) [inline atomic_reactor/schemas/config.json:794]: Why is this needed as part of this change?

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 3 above)
Comment (SalatskySal) [reply atomic_reactor/plugins/pre_reactor_config.py:481]: Some tests are using 'DEFAULT_CONFIG' to configure reactor_config_map. 
Inasmuch as 'koji' is mandatory, I've decided to add 'koji' into 'DEFAULT_CONFIG'.

File: atomic_reactor/plugins/pre_resolve_composes.py (hunk 4 above)
Comment (SalatskySal) [reply atomic_reactor/plugins/pre_resolve_composes.py:76]: My mistake, I'll fix that

File: atomic_reactor/plugins/exit_store_metadata_in_osv3.py (hunk 2 above)
Comment (SalatskySal) [reply atomic_reactor/plugins/exit_store_metadata_in_osv3.py:179]: Even if we delete empty dict, it won't change the behavior of this function, because of 'koji' will always be in reactor_config_map('koji' is required in the schema).

File: atomic_reactor/plugins/pre_resolve_composes.py (hunk 4 above)
Comment (SalatskySal) [reply atomic_reactor/plugins/pre_resolve_composes.py]: Fixed

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 3 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_reactor_config.py:481]: Fix tests, don't modify code with random default values because tests are failing

Comment (SalatskySal) [reply atomic_reactor/plugins/pre_reactor_config.py]: Fixed

File: tests/plugins/test_download_remote_source.py [hunk 6]
```diff
@@ -36,12 +36,16 @@
...
+            hub_url: /
+            root_url: ''
+            auth: {}
         cachito:
            api_url: 'example.com'
            insecure: {}
            auth:
                ssl_certs_dir: /some/dir
-        """.format(insecure))
+        """.format("{}", insecure))
```
Comment (chmeliik) [inline tests/plugins/test_download_remote_source.py:48]: When you need a literal `{}` in a formatted string, you can use this:
```python
"a: {{}}, b: {}".format(1)  # -> "a: {}, b: 1"
```
IMO it is more readable.

Note that this appears in other places as well.

Comment (SalatskySal) [reply tests/plugins/test_download_remote_source.py:48]: Wow, thanks :+1: :+1: :+1:

Comment (SalatskySal) [reply tests/plugins/test_download_remote_source.py]: Fixed

**Comment by MartinBasti** [comment]
> @SalatskySal @MartinBasti I think that also all affected plugins, should now require reactor_config_map, and not have it optional, like most of them do

can you elaborate more what you mean?

**Comment by rcerven** [comment]
> > @SalatskySal @MartinBasti I think that also all affected plugins, should now require reactor_config_map, and not have it optional, like most of them do
> 
> can you elaborate more what you mean?

all affected plugins should do this cleanup :
https://github.com/containerbuildsystem/atomic-reactor/pull/1448/files#diff-32d320c81fb4648113f8cd623cf0caf1L383

removing reactor_config param from tests and ALWAYS create one

**Comment by MartinBasti** [comment]
Can we merge this and do it in another PR?

**Comment by rcerven** [comment]
> Can we merge this and do it in another PR?

well he is doing it already here for some plugins, and it is clear from PR which plugins are affected, but up to you

**Comment by SalatskySal** [comment]
> > Can we merge this and do it in another PR?
> 
> well he is doing it already here for some plugins, and it is clear from PR which plugins are affected, but up to you

@rcerven affected plugins have been updated

**Comment by rcerven** [comment]
> > > Can we merge this and do it in another PR?
> > 
> > 
> > well he is doing it already here for some plugins, and it is clear from PR which plugins are affected, but up to you
> 
> @rcerven affected plugins have been updated

looks good, only 1 missing now test_tag_and_push

**Comment by MartinBasti** [comment]
Release notes:
 - Removed deprecated koji fallback methods (`koji` configuration is mandatory now)
